src/Twilio/Http/Request.cs

```diff
@@ -238,8 +258,12 @@ public override bool Equals(object obj)
             var other = (Request)obj;
             return Method.Equals(other.Method) &&
                    buildUri().Equals(other.buildUri()) &&
-                   QueryParams.All(other.QueryParams.Contains) &&
-                   PostParams.All(other.PostParams.Contains);
+                   QueryParams.All(q => other.QueryParams.Contains(q)) &&
+                   other.QueryParams.All(qOther => QueryParams.Contains(qOther)) &&
+                   PostParams.All(p => other.PostParams.Contains(p)) &&
+                   other.PostParams.All(pOther => PostParams.Contains(pOther)) &&
+                   HeaderParams.All(h => other.HeaderParams.Contains(h)) &&
+                   other.HeaderParams.All(hOther => HeaderParams.Contains(hOther));
```

Comment:
Oh, nice catch. But can the old syntax still be used?

Reply:
Not sure, will test and update if it can